Lab1_Operationalizing_Pytorch_with_Mlflow/README.md

```diff
@@ -1,31 +1,127 @@
+web access to the models.
+
+In order to provide feedback loop, a **Monitor Node** checks the status and performance meters, issuing 
+new training cycles or a full rebuild process. Some structures traverses through Development and Production perimeters. Training and Monitor nodes are key elements that communicate between those zones. 
+**Control or Operator Machine** is the VM used by Configuration Manager or Operation Engineer to follow up
+the process or issue commands to the nodes in the pipeline. This process may run through many different 
+perimeters, however this solution comprises _Development_ and _Production_ stages only.
+
+### Architectural Components and Decisions
+
+In this section, we record the (most relevant) design decisions, components selection and the rationale around them.
+
+| # | Subject | Description | Affected Items | Solution | Rationale |
+|---|---------|-------------|----------------|----------|-----------|
+| 1 | Computing | ML needs processing power as well as GPU. Our architecure relies on distributed elements, thus needing a reasonable number of physical or virtual machines (some with GPU-power) | All Nodes and Machines | GCP Compute Service | It provides $ 300,00 of (enough) free services and we can also spin some gpu-powered machines |
+| 2 | Durable Data Storage | Every serious solution needs to store durable data | Tracking Server | Google Cloud SQL | A Postgres Database hosted by Google Cloud SQL is reliable and fast enough for our needs  |
+| 3 | Large and Binary Data Storage | ML projects eventually need to store models and extra large files, This type of data doesn't fit in RDBMS | Tracking Server<br>Artifact Store | Google Cloud Storage | Same as #1 |
+| 4 | Shipment and Deployment | There are many deployable assets as well as acessory tools that need specific environment and SO libraries to run. Just installing them into a machine isn't a viable option | Training  Node<br>Google Cloud Services | Docker | It's the _de facto_ pattern for shipping and deploying things |
+| 5 | Lib Management | Python has a powerful but sometimes confusing and conflicting library ecossystem. Our components may depend on conflicting libs and that will lead to problems in Production | All Nodes and Machines | Conda<br>Pip | Both are heavily used by Python community. They also work quite well with isolation provided by Docker images |
+| 6 | Remote Access | Sometimes we need to issue remote commands or operate a remote machine | All Nodes and Machines | SSH | Hey... It's SSH. There's no need to rationale :-P |
+| 7 | CI/CD & Workflow Management | We need to seamlessly integrate and deploy ML stuff. We also have to coordinate the process between nodes | All Items | Github Actions | We're using GitHub, it's heavily used by community and comes off-the-shelf |
+| 8 | Scheduler | Some activities are triggered by clock | All Nodes and Machines | CRON | We are using Unix like machines |
```

Comment:
Just wondering what we use cron for

Reply:
Nothing. Forgot it there. We thought it would be of use, but we can do everything with GH + gcloud.